### DISCUSSION TITLE
Use payloadcms with a frontend rendering like Gatsby?

### DISCUSSION THREAD

**Discussion by Jarrodsz** [payloadcms/payload#1155] [Q&A]
Edit: rephrasing my question.

The payloadcms.com website is a usable website to use as starters assuming it is also build with payloadcms my question would be is it opensourced? Like gatsby have there website on github it was a big help to see how things work in a real live example.

Instead of reinventing the wheel would it be possible to provide a more mature repo or the source of payloadcms.com site to get a "full" "starter theme" instead of having to reinvent the wheel?

Perhaps a build with payloadcms section would be helpfull where one coule list there site with optional github repo if its opensource. Gatsby has this.

Since gatsby is higly optimized how does a nextjs application is compared to nextjs with payload?

Gatsby is not a requirement for us but currently all my landing pages are build with it. Would it be possible to even use gaysby with payloadcms?

Kind regards

**Comment by DanRibbens** [comment]
We are not prepared to open-source our website, but you can get a good example from our boilerplates or if you want something more built up to use check out the repo from this video series about building professional sites:

Repo - https://github.com/payloadcms/custom-website-series
Video - https://www.youtube.com/watch?v=bxWsZTtqs80&t=476s

I haven't seen a boilerplate with Gatsby + Payload yet, but those two should work well together if you wanted to make one.

**Reply by Jarrodsz** [reply]
> We are not prepared to open-source our website, but you can get a good example from our boilerplates or if you want something more built up to use check out the repo from this video series about building professional sites:
> 
> Repo - https://github.com/payloadcms/custom-website-series Video - https://www.youtube.com/watch?v=bxWsZTtqs80&t=476s
> 
> I haven't seen a boilerplate with Gatsby + Payload yet, but those two should work well together if you wanted to make one. The provided repro is bare bones missing most essential parts that any modern website needs. So it's basically not usable as a starter and needs many time to add additional features that are already written ( but not shared ) at the moment. 

May I ask why not? your site has implemented al the features a modern website needs.
Why would you let your user base reinvent the wheel following all kinds of tutorials when a " full site example " is already implemented in the payload cms website?

Im not saying I request a full copy of your website, just an better "site starter" based upon it with the regions content and data model implemented so one can easily start there own website of payload.

I don't have the time to invest hours in the cms to get something rendered in the end, that I can do now with a normal Gatsby site myself using react components ( sure I don't have edit capabilities ) but since its such a small site and im the only author its more time efficient.

Can the core team please consider providing a "full site" example that is mature like the features that are on payloadcms website so we don't have to invest hours in reinventing the wheel?

It would also largely improve the adoption of payload cms its the most promising cms development in its ecosphere. 

If this issue would be resolved and you provide multi tenant example / setup / better docs on how to do it 
this cms is gamechanger.

**Reply by denolfe** [reply]
We definitely hear the need for more starters and boilerplates. That is going to be a focus for us in the coming weeks.

I also have great news: we will be fully rebuilding our website in public over the next few weeks. We'll be announcing this very soon, but in the meantime, you can follow along on the repo here: https://github.com/payloadcms/website

**Comment by thompsonsj** [comment]
@Jarrodsz I've been using Payload CMS with Gatsby for a while but only with standard API calls.

Now I've just started work on a Gatsby source plugin for Payload CMS:

https://github.com/thompsonsj/gatbsy-source-payload-cms

At the time of writing, I'm still to figure out the best way to handle assets/images, but hey - it's working and it seems to be a good way of getting up and running with Payload CMS and Gatsby!
